src/org/labkey/test/tests/AdminConsoleNavigationTest.java

```diff
@@ -71,7 +108,134 @@ public void testAdminNavTrails()
+                "Running Threads"      // Undesired consequences
+        ));
+        ShowAdminPage adminConsole = goToAdminConsole();
+        List<WebElement> adminLinks = adminConsole.getAllAdminConsoleLinks();
+        Map<String, String> linkHrefs = new LinkedHashMap<>();
+        for (WebElement link : adminLinks)
+            linkHrefs.put(link.getText(), link.getAttribute("href"));
+
+        log("Verifying links can be accessed by admin");
+        for (Map.Entry<String, String> link : linkHrefs.entrySet())
+        {
+            if (ignoredLinksAdmin.contains(link.getKey()))
+                TestLogger.log("Skipping admin link: " + link.getKey());
+            else
+            {
+                log("Verifying link " + link.getKey() + " with URL " + link.getValue());
+                verifyLinks(link.getKey(), link.getValue(), 200);
+            }
+        }
+
+        log("Verifying links cannot be accessed by non admin");
+        goToHome();
+        impersonate(NON_ADMIN);
+        for (Map.Entry<String, String> link : linkHrefs.entrySet())
+        {
+            if (ignoredLinksNonAdmin.contains(link.getKey()))
+                TestLogger.log("Skipping admin link: " + link.getKey());
+            else
+            {
+                log("Verifying link " + link.getKey() + " with URL " + link.getValue());
+                verifyLinks(link.getKey(), link.getValue(), 403);
+            }
+        }
+        goToHome();
+        stopImpersonating();
+    }
+
+    private void verifyReadOnlyLinks(String text, String url)
+    {
+        beginAt(url);
+        Assert.assertEquals("URL " + url + " is broken for " + text, 200, getResponseCode());
+        Locator.XPathLocator updateButtonLoc = Locator.XPathLocator.union(
+                Locator.linkWithText("Submit"), Locator.linkWithText("Save"), Locator.linkWithText("Update")).notHidden();
+        assertFalse("Either Save/Submit/Update button exists in the page " + text, isElementPresent(updateButtonLoc));
+    }
+
+    private void verifyPostCommandFails(String url)
+    {
+        String[] actionController = url.substring(url.lastIndexOf('/') + 1).split("-");
+        Connection connection = WebTestHelper.getRemoteApiConnection();
+        SimplePostCommand command = new SimplePostCommand(actionController[0], actionController[1]);
+        try
+        {
+            command.execute(connection, getCurrentContainerPath());
+        }
+        catch (IOException e)
+        {
+            throw new RuntimeException(e);
+        }
+        catch (CommandException e)
+        {
+            Assert.assertEquals("Post command should not go through", 404, e.getStatusCode());
+        }
+        catch (JavascriptException e)
+        {
+            return; //Do nothing for admin-showAllErrors.view for error LABKEY is not defined
+        }
+    }
+
+    private void verifyLinks(String text, String url, int expectedResponseCode)
```

Comment:
Only verifies a single link
```suggestion
    private void verifyLink(String text, String url, int expectedResponseCode)
```

Reply:
Done